src/index.ts

```diff
@@ -65,16 +65,20 @@ async function uploadStateOnFinalized(): Promise<void> {
         storedWSEpoch = await ipfsApiClient.getIPFSWSEpoch(CID);
       }
 
-      console.log("Weak subjectivty epoch from IPNS: ", storedWSEpoch);
-      console.log("Weak subjectivty epoch from beacon node: ", wsEpoch);
+      console.log("Weak subjectivity epoch from IPNS: ", storedWSEpoch);
+      console.log("Weak subjectivity epoch from beacon node: ", wsEpoch);
 
       if (wsEpoch > storedWSEpoch) {
-        alreadyFetchingState = true;
         try {
+          alreadyFetchingState = true;
           await getAndUploadState(wsEpoch);
+          alreadyFetchingState = false;
         } catch (error) {
           console.error(error.message);
-          setTimeout(async () => await getAndUploadState(wsEpoch), 5000);
+          setTimeout(async () => {
+            await getAndUploadState(wsEpoch);
+            alreadyFetchingState = false;
+          }, 5000);
         }
```

Comment:
This is not error handling safe. If you need to implement a retry functionality please use https://www.npmjs.com/package/async-retry